### ISSUE TITLE
AbstractComponent.setCaption() doesn't work in combination with CustomLayout

### ISSUE THREAD

**Issue by bc15** [vaadin/framework#8413]
TextField.setCaption() doesn't work in combination with CustomLayout.

This means the user interface doesnt't get refreshed.
It works when I wrap the TextField in an VerticalLayout.

I tested with vaadin 7.3.5 and vaadin 7.3.6, see the following minimal example.
When I click the button in the example, only the second example gets refreshed.

````
@Theme("test")
public class MyVaadinUI extends UI
{
    @Override
    protected void init(VaadinRequest vaadinRequest)
    {
        CustomLayout content = new CustomLayout("layoutname");
        content.setSizeUndefined();
        setContent(content);

        Button loginButton = new Button("Test");
        TextField username1 = new TextField();
        TextField username2 = new TextField();
        username1.setCaption("doesn't work");
        username2.setCaption("doesn't work");
        content.addComponent(username1, "test1");
        content.addComponent(new VerticalLayout(username2), "test2");
        content.addComponent(loginButton, "okbutton");

        loginButton.addClickListener(e -> {
            username1.setCaption("works");
            username2.setCaption("works");
        });
    }
}
````
VAADIN/themes/test/layouts/layoutname.html
````
<table width="100%" height="100%">
  <tr height="100%">
    <td>
      <table align="center">
        <tr>
          <td><div location="test1"></div></td>
        </tr>
        <tr>
          <td><div location="test2"></div></td>
        </tr>
      </table>
    </td>
  </tr>
  <tr>
    <td align="right" colspan="2">
      <div location="okbutton"></div>
    </td>
  </tr>
</table>
````

**Comment by bc15** [comment]
This doesn't work in vaadin 8.0.0 anymore.
I confirm it works in 7.7.7

**Comment by pleku** [comment]
As mentioned in #8413, the fixes, including this one, are being picked from 7.7.7 to 8.0 currently (#8300), and will be released in 8.0.1.
